### PR TITLE
Feature/fixedpoint trig functions

### LINKED ISSUE #927: implement ATan2

### LINKED ISSUE #989: Implement trigonometry functions for FixedPoint

### PR DESCRIPTION
**Pull request by markos**
Sin(), Cos() implemented for FixedPoint
added Round()
various cosmetic fixes, renames, etc
Added lots of testcases for Sin/Cos.
Added template specializations for FixedPoint for all trig functions
Implement ATan2 specialization.

Closes: #927 
Closes: #989

### CODE REVIEW

File: libs/math/include/math/fixed_point/fixed_point.hpp [hunk 1]
```diff
@@ -828,24 +842,95 @@ class FixedPoint : public BaseFixedpointType
...
+    quadrant_funcs.push_back([](FixedPoint const &x) { return -SinPi2(x); });
+    quadrant_funcs.push_back([](FixedPoint const &x) { return -CosPi2(x); });
+    quadrant_funcs.push_back([](FixedPoint const &x) { return SinPi2(x); });
+
+    return quadrant_funcs[(size_t)quadrant](r - CONST_PI_2 * quadrant);
+  }
+
+  static constexpr FixedPoint Tan(FixedPoint const &x)
+  {
+    return x;
```
Comment (baykaner) [inline libs/math/include/math/fixed_point/fixed_point.hpp]: I don't think this implementation is finished yet ;-)
If you want to merge in this PR before finishing this implementation, why don't you throw std::runtime_error("Tan not implemented yet")

File: libs/math/tests/math/fixed_point/fixed_point.cpp [hunk 2]
```diff
@@ -999,3 +1101,275 @@ TEST(FixedPointTest, SQRT_32_32)
...
+
+  fp64 step{0.1};
+  fp64 x{-fp64::CONST_PI * 100};
+  for (; x < fp64::CONST_PI * 100; x += step)
+  {
+    fp64 e = fp64::Cos(x);
+    delta  = (double)e - std::cos((double)(x));
+    EXPECT_NEAR(delta, 0.0, (double)fp64::TOLERANCE);
+  }
+}
```
Comment (baykaner) [inline libs/math/tests/math/fixed_point/fixed_point.cpp:1373]: nice!

would be neat to add a couple of benchmarks for the trig functions too

**Review by baykaner — CHANGES_REQUESTED** [review]
looks pretty good as far as I can tell;
only concerns are the Tan non-implementation should be guarded, and it would be nice to have benchmarks for the more costly trig functions

File: libs/math/include/math/fixed_point/fixed_point.hpp (hunk 1 above)
Comment (markos) [reply libs/math/include/math/fixed_point/fixed_point.hpp]: I added a trivial implementation, however a proper and performant implementation will need handling of -/+ infinity to work properly, and will likely need to come at the same time as ATan().

File: libs/math/tests/math/fixed_point/fixed_point.cpp (hunk 2 above)
Comment (markos) [reply libs/math/tests/math/fixed_point/fixed_point.cpp:1373]: you mean in this PR? :)

Comment (markos) [reply libs/math/tests/math/fixed_point/fixed_point.cpp:1373]: I'll start writing a benchmark for FixedPoint functions and will include this in a separate PR.

**Pull request closed without merging**